src/app/src/main/kotlin/org/imperial/mrc/hint/controllers/CalibrateController.kt

```diff
@@ -0,0 +1,67 @@
+package org.imperial.mrc.hint.controllers
+
+import org.springframework.web.bind.annotation.*
+import org.springframework.http.ResponseEntity
+import org.imperial.mrc.hint.db.CalibrateDataRepository
+import org.imperial.mrc.hint.clients.HintrAPIClient
+import org.imperial.mrc.hint.models.ModelOptions
+import org.imperial.mrc.hint.models.SuccessResponse
+import org.imperial.mrc.hint.models.asResponseEntity
+import org.imperial.mrc.hint.exceptions.CalibrateDataException
+import com.fasterxml.jackson.databind.ObjectMapper
+import org.springframework.http.HttpStatus
+
+@RestController
+@RequestMapping("/calibrate")
+class CalibrateController(val apiClient: HintrAPIClient, val calibrateDataRepository: CalibrateDataRepository)
+{
+    @GetMapping("/options/{iso3}")
+    @ResponseBody
+    fun calibrationOptions(@PathVariable("iso3") iso3: String): ResponseEntity<String>
+    {
+        return apiClient.getModelCalibrationOptions(iso3)
+    }
+
+    @PostMapping("/submit/{id}")
+    @ResponseBody
+    fun calibrateSubmit(@PathVariable("id") runId: String, @RequestBody modelCalibrateOptions: ModelOptions):
+            ResponseEntity<String>
+    {
+        return apiClient.calibrateSubmit(runId, modelCalibrateOptions)
+    }
+
+    @GetMapping("/status/{id}")
+    @ResponseBody
+    fun calibrateStatus(@PathVariable("id") id: String): ResponseEntity<String>
+    {
+        return apiClient.getCalibrateStatus(id)
+    }
+
+    @GetMapping("/result/metadata/{id}")
+    @ResponseBody
+    fun calibrateResultMetadata(@PathVariable("id") id: String): ResponseEntity<String>
+    {
+        return apiClient.getCalibrateResultMetadata(id)
+    }
+
+    @GetMapping("/result/data/{id}")
+    @ResponseBody
+    fun calibrateResultData(@PathVariable("id") id: String): ResponseEntity<String>
+    {
+        val res = apiClient.getCalibrateResultData(id)
+        if (res.statusCode != HttpStatus.OK) {
+            throw CalibrateDataException("Failed to fetch result")
+        }
+        val jsonBody = ObjectMapper().readTree(res.body?.toString())
+        val path = jsonBody.get("data").get("path").textValue()
+        val resObj = calibrateDataRepository.getDataFromPath(path)
```

Comment:
These should be handled in service. You also don't need throwing exception manually. This gets intercepted in the global ExceptionHandler.

Reply:
can move it to a service, also is there a place where you have defined it to throw string "failed to fetch results" or will it just log out the full error

Reply:
done and have removed exception for now

Reply:
We want to translate the error and not allow the raw message through.

Reply:
that would be nice to have, currently itll just give the generic "an unexpected error occurred..." error message, would be nice to improve these for the calibrate data stuff at the very least, is this another ticket I hear?

Reply:
Actually, the generic message would be fine to send back to the user, but we should look into logging the original message in the back end - I think you might have already made a ticket for that?